src/identity.jl

```diff
@@ -101,21 +103,52 @@ function Base.:(*)(A::AbstractMatrix, 𝐼::IdentityMultiple)
     return A * 𝐼.M.λ
 end
 
+# right-division
+function Base.:(/)(A::AbstractMatrix, 𝐼::IdentityMultiple)
+    @assert size(A, 2) == 𝐼.n
+    return A * inv(𝐼.M.λ)
+end
+
 function Base.:(+)(𝐼1::IdentityMultiple, 𝐼2::IdentityMultiple)
     @assert 𝐼1.n == 𝐼2.n
     return IdentityMultiple(𝐼1.M + 𝐼2.M, 𝐼1.n)
 end
 
+function Base.:(-)(𝐼1::IdentityMultiple, 𝐼2::IdentityMultiple)
+    @assert 𝐼1.n == 𝐼2.n
+    return IdentityMultiple(𝐼1.M - 𝐼2.M, 𝐼1.n)
+end
+
 function Base.:(*)(𝐼1::IdentityMultiple, 𝐼2::IdentityMultiple)
     @assert 𝐼1.n == 𝐼2.n
     return IdentityMultiple(𝐼1.M * 𝐼2.M, 𝐼1.n)
 end
 
+function Base.:(*)(𝐼::IdentityMultiple{T}, U::UniformScaling{S}) where {T<:Number, S<:Number}
+    return IdentityMultiple(𝐼.M.λ * U, 𝐼.n)
+end
+
+function Base.:(*)(U::UniformScaling{T}, 𝐼::IdentityMultiple{S}) where {T<:Number, S<:Number}
+    return IdentityMultiple(𝐼.M.λ * U, 𝐼.n)
+end
+
+function Base.:(/)(𝐼::IdentityMultiple{T}, U::UniformScaling{S}) where {T<:Number, S<:Number}
+    @assert !iszero(U.λ)
+    return IdentityMultiple(𝐼.M * inv(U.λ), 𝐼.n)
+end
+
+function Base.:(/)(U::UniformScaling{T}, 𝐼::IdentityMultiple{S}) where {T<:Number, S<:Number}
+    @assert !iszero(𝐼.M.λ)
+    return IdentityMultiple(U * inv(𝐼.M.λ), 𝐼.n)
+end
+
 function Base.show(io::IO, ::MIME"text/plain", 𝐼::IdentityMultiple{T}) where T
     print(io, "$(typeof(𝐼)) of value $(𝐼.M.λ) and order $(𝐼.n)")
 end
 
-# callable identity matrix
+# callable identity matrix given the size and the numeric type
 LinearAlgebra.I(n::Int, N=Float64) = IdentityMultiple(one(N)*I, n)
```

Comment:
```suggestion
 LinearAlgebra.I(n::Int, N::DataType=Float64) = IdentityMultiple(one(N)*I, n)
```
to avoid that `I(3,3)` throws a ambiguity error and correct behavior of `I(3,3.0)`

Reply:
Good catch, thanks.